harness/determined/common/api/authentication.py

```diff
@@ -34,18 +34,12 @@ def get_det_password_from_env() -> Optional[str]:
     return os.environ.get("DET_PASS")
 
 
-class UsernameTokenPair:
-    def __init__(self, username: str, token: str):
-        self.username = username
-        self.token = token
-
-
 def login(
     master_address: str,
     username: str,
     password: str,
     cert: Optional[certs.Cert] = None,
-) -> UsernameTokenPair:
+) -> "api.Session":
```

Comment:
Because `harness/determined/common/api/_session.py` imports `harness/determined/common/api/__init__.py`, which imports `harness/determined/common/api/authentication.py`. Quoting `api.Session` here keeps the compiler from trying to resolve the circle.

This isn't needed anymore in Python 3.X.Y, where X.Y > 8.5, I think. But this is just the vaguest memory and I don't know.

Reply:
https://peps.python.org/pep-0484/#forward-references and 
```python
from __future__ import annotations
```
which became available in 3.7. Our current codebase prefers quoting those forward references, but it looks like the alternative is available to us now, too.

Reply:
i see. i generally view circular dependencies as a sign of poor design/separation of concerns. but in this case i'm not sure what else you'd do aside from refactoring everything, so this seems like a fine solution